### ISSUE TITLE
Print healthcheck-nodeport annotation message only for LoadBalancer type services

### ISSUE THREAD

**Issue by bonifaido** [kubernetes/kubernetes#42888]
---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): BUG REPORT

**Kubernetes version** (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.4", GitCommit:"7243c69eb523aa4377bce883e7c0dd76b84709a1", GitTreeState:"clean", BuildDate:"2017-03-08T02:50:34Z", GoVersion:"go1.8", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.2", GitCommit:"08e099554f3c31f6e6f07b448ab3ed78d0520507", GitTreeState:"clean", BuildDate:"2017-01-12T04:52:34Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:
- **Cloud provider or hardware configuration**: AWS m4.2xlarge
- **OS** (e.g. from /etc/os-release): Ubuntu 16.04
- **Kernel** (e.g. `uname -a`): 4.4.0-36-generic
- **Install tools**: kubeadm 
- **Others**: 1 master and 1 minion

**What happened**:
This [guide](https://kubernetes.io/docs/tutorials/services/source-ip/#source-ip-for-services-with-typenodeport) states that if using `service.beta.kubernetes.io/external-traffic: OnlyLocal` kube-proxy will only proxy requests to local endpoints, never forwarding traffic to other nodes and thereby preserving the original source IP address. However when I annotate my NodePort type service with this annotation I see the following error in the logs of kube-proxy of the node where the only pod of the service is running:
```
E0310 13:57:12.875026       1 proxier.go:488] Service does not contain necessary annotation service.beta.kubernetes.io/healthcheck-nodeport
```
After this message the service stops working outside of the cluster on the specified NodePort.

**What you expected to happen**:
I expected to see that only the with the pod on it serves traffic.

**How to reproduce it** (as minimally and precisely as possible):
```
# 1. Create the Deployment and expose it on NodePort service
kubectl run python --port 8000 --image python -- python -m http.server
kubectl expose deployment python --port 8000 --type NodePort
# Expect the usual directory listing here of the builtin Python webserver.
curl $MINION:$NODEPORT

# 2. Annotate the service with OnlyLocal
kubectl annotate service python service.beta.kubernetes.io/external-traffic=OnlyLocal
# This call will hang for 30 seconds than exit.
curl $MINION:$NODEPORT
```

**Anything else we need to know**:

**Comment by bonifaido** [comment]
Sorry, I have found the issue, I had a setup where the none of the node names were the same as the hostname of the kube-proxy pod and this caused the issue, where none of the Pods running on a hosts were recognised as local pods. I have fixed this issue by manually setting the hostname with `--hostname-override` in kube-proxy.
However I still insist that this error message is misleading and shouldn't be printed in case of NodePort type services, but only for just LoadBalancers.
